nkms/characters.py

```diff
@@ -253,44 +243,41 @@ def verify_from(self,
 
         return is_valid, cleartext
 
-    def decrypt(self, message_kit):
-        return self._crypto_power.decrypt(message_kit)
+    """
+    Next we have decrypt() and sign() - these two functions use the private keys of their respective powers;
+    any character who has these powers can use these functions.
 
-    def _lookup_actor(self, actor: "Character"):
-        try:
-            return self._actor_mapping[actor.id()]
-        except KeyError:
-            raise self.NotFound(
-                "We haven't learned of an actor with ID {}".format(actor.id()))
+    If they don't have the correct Power, the appropriate PowerUpError is raised.
+    """
+
+    def decrypt(self, message_kit):
+        return self._crypto_power.power_ups(EncryptingPower).decrypt(message_kit)
 
-    def id(self):
-        return hexlify(bytes(self.seal))
+    def sign(self, message):
+        return self._crypto_power.power_ups(SigningPower).sign(message)
 
-    def public_key(self, key_class):
-        # TODO: Does it make sense to have a specialized exception here? Probably.
-        return self._crypto_power.public_keys[key_class]
+    def public_key(self, power_up_class):
+        power_up = self._crypto_power.power_ups(power_up_class)
+        return power_up.public_key()
 
 
 class Alice(Character):
     _server_class = NuCypherSeedOnlyDHTServer
     _default_crypto_powerups = [SigningPower, EncryptingPower]
 
-    def generate_kfrags(self, bob, m, n):
+    def generate_kfrags(self, bob, m, n) -> List:
```

Comment:
I feel like we can give this a better name than `generate_kfrags` as that doesn't really sound like it's doing anything like `split_rekey`. However, it's too late for me to offer a solution at this time. I'll think about this and come back to you.